dex/farm/src/base_functions.rs

```diff
@@ -99,25 +104,56 @@ pub trait BaseFunctionsModule:
                 &base_exit_farm_result.storage_cache.reward_token_id,
             );
         }
-
         (farming_token_payment, reward_payment).into()
     }
 
+    fn merge_farm_tokens_with_energy(&self, caller: &ManagedAddress) -> EsdtTokenPayment<Self::Api> {
```

Comment:
This should not be a separate method at all. It's basically a copy of the normal merging function, with a call to update energy added at the end.

Reply:
Deleted.